test/correctness/simd_op_check_hvx.cpp

```diff
@@ -0,0 +1,664 @@
+        check("v*.w += vrmpy(v*.b,v*.b)", hvx_width, i32_1 + i32(i8_1)*i8_1 + i32(i8_2)*i8_2 + i32(i8_3)*i8_3 + i32(i8_4)*i8_4);
+
+#if 0
+        // These don't generate yet because we don't support mixed signs yet.
+        check("vrmpy(v*.ub,v*.b)", hvx_width, i32(u8_1)*i8_1 + i32(u8_2)*i8_2 + i32(u8_3)*i8_3 + i32(u8_4)*i8_4);
+        check("v*.w += vrmpy(v*.ub,v*.b)", hvx_width, i32_1 + i32(u8_1)*i8_1 + i32(u8_2)*i8_2 + i32(u8_3)*i8_3 + i32(u8_4)*i8_4);
+        check("vrmpy(v*.ub,v*.b)", hvx_width, i16(u8_1)*i8_1 + i16(u8_2)*i8_2 + i16(u8_3)*i8_3 + i16(u8_4)*i8_4);
+#endif
+        // Temporarily disabling this vrmpy test because of https://github.com/halide/Halide/issues/4248
+        // These should also work with 16 bit results. However, it is
+        // only profitable to do so if the interleave simplifies away.
+        // Expr u8_4x4[] = {
+        //     in_u8(4*x + 0),
+        //     in_u8(4*x + 1),
+        //     in_u8(4*x + 2),
+        //     in_u8(4*x + 3),
+        // };
+        // Disabling this test right now
+        // check("vrmpy(v*.ub,r*.b)", hvx_width/2, i16(u8_4x4[0])*127 + i16(u8_4x4[1])*126 + i16(u8_4x4[2])*-125 + i16(u8_4x4[3])*124);
+
+        // Make sure it doesn't generate if the operands don't interleave.
+        check("vmpa(v*.ub,r*.b)", hvx_width, i16(u8_1)*127 + i16(u8_2)*-126 + i16(u8_3)*125 + i16(u8_4)*124);
+
+        check("v*.w += vasl(v*.w,r*)", hvx_width/4, u32_1 + (u32_2 * 8));
+        check("v*.w += vasl(v*.w,r*)", hvx_width/4, i32_1 + (i32_2 * 8));
+        check("v*.w += vasr(v*.w,r*)", hvx_width/4, i32_1 + (i32_2 / 8));
+
+        check("v*.w += vasl(v*.w,r*)", hvx_width/4, i32_1 + (i32_2 << u32(y % 32)));
+        check("v*.w += vasr(v*.w,r*)", hvx_width/4, i32_1 + (i32_2 >> u32(y % 32)));
+
+        if (isa_version >= 65) {
+            check("v*.h += vasl(v*.h,r*)", hvx_width/2, i16_1 + (i16_2 << u16(y % 16)));
+            check("v*.h += vasl(v*.h,r*)", hvx_width/2, i16_1 + (i16(y % 16) << u16_2));
+            check("v*.h += vasr(v*.h,r*)", hvx_width/2, i16_1 + (i16_2 >> u16(y % 16)));
+            check("v*.h += vasl(v*.h,r*)", hvx_width/2, u16_1 + (u16_2 * 16));
+            check("v*.h += vasl(v*.h,r*)", hvx_width/2, i16_1 + (i16_2 * 16));
+            check("v*.h += vasl(v*.h,r*)", hvx_width/2, u16_1 + (16 * u16_2));
+            check("v*.h += vasl(v*.h,r*)", hvx_width/2, i16_1 + (16 * i16_2));
+            check("v*.h += vasr(v*.h,r*)", hvx_width/2, i16_1 + (i16_2 / 16));
+        }
+
+        check("vcl0(v*.uh)", hvx_width/2, count_leading_zeros(u16_1));
+        check("vcl0(v*.uw)", hvx_width/4, count_leading_zeros(u32_1));
+        check("vnormamt(v*.h)", hvx_width/2, max(count_leading_zeros(i16_1), count_leading_zeros(~i16_1)));
+        check("vnormamt(v*.w)", hvx_width/4, max(count_leading_zeros(i32_1), count_leading_zeros(~i32_1)));
+        check("vpopcount(v*.h)", hvx_width/2, popcount(u16_1));
+    }
+};
+
+int main(int argc, char **argv) {
+
```

Comment:
No blank line between opening brace and first statement in a function

Reply:
Done